addons/account_peppol/tests/test_peppol_messages.py

```diff
@@ -0,0 +1,463 @@
+            'ref': 'Test reference',
+            'invoice_line_ids': [
+                (0, 0, {
+                    'name': 'line 1',
+                    'product_id': self.product_a.id,
+                }),
+                (0, 0, {
+                    'name': 'line 2',
+                    'product_id': self.product_a.id,
+                }),
+            ],
+        })
+
+    def create_invoice_send_wizard(self, moves, **wizard_kwargs):
+        action = moves.action_send_and_print()
+        action_context = action['context']
+        return self.env['account.invoice.send'].with_context(
+            action_context,
+            active_ids=moves.ids
+        ).create({'is_print': False, **wizard_kwargs})
+
+    @classmethod
+    def _get_mock_data(cls, error=False):
+        proxy_documents = {
+            FAKE_UUID[0]: {
+                'accounting_supplier_party': False,
+                'filename': 'test_outgoing.xml',
+                'enc_key': '',
+                'document': '',
+                'state': 'done' if not error else 'error',
+                'direction': 'outgoing',
+                'document_type': 'Invoice',
+            },
+            FAKE_UUID[1]: {
+                'accounting_supplier_party': '0198:dk16356706',
+                'filename': 'test_incoming',
+                'enc_key': file_open(f'{FILE_PATH}/enc_key', mode='rb').read(),
+                'document': b64encode(file_open(f'{FILE_PATH}/document', mode='rb').read()),
+                'state': 'done' if not error else 'error',
+                'direction': 'incoming',
+                'document_type': 'Invoice',
+            },
+        }
+
+        responses = {
+            '/api/peppol/1/send_document': {'result': {
+                'messages': [{'message_uuid': FAKE_UUID[0]}]}},
+            '/api/peppol/1/ack': {'result': {}},
+            '/api/peppol/1/get_all_documents': {'result': {
+                'messages': [
+                    {
+                        'accounting_supplier_party': '0198:dk16356706',
+                        'filename': 'test_incoming.xml',
+                        'uuid': FAKE_UUID[1],
+                        'state': 'done',
+                        'direction': 'incoming',
+                        'document_type': 'Invoice',
+                        'sender': '0198:dk16356706',
+                        'receiver': '0208:0477472701',
+                        'timestamp': '2022-12-30',
+                        'error': False if not error else 'Test error',
+                    }
+                ],
+            }}
+        }
+        return proxy_documents, responses
+
+    @contextmanager
+    def _set_context(self, other_context):
+        previous_context = self.env.context
+        self.env.context = dict(previous_context, **other_context)
+        yield self
+        self.env.context = previous_context
+
+    @classmethod
+    def _request_handler(cls, s: Session, r: PreparedRequest, /, **kw):
```

Comment:
This breaks installing account_peppol in 16 for the minimal Python version, which is 3.7
https://github.com/odoo/odoo/blob/6f40f3a85d4c6edf43a427d8575c5c95c3005cd4/odoo/__init__.py#L19
```suggestion
    def _request_handler(cls, s: Session, r: PreparedRequest, **kw):
```